pytorch_lightning/core/lightning.py

```diff
@@ -403,6 +409,31 @@ def log_dict(
                 add_dataloader_idx=add_dataloader_idx
             )
 
+    @staticmethod
+    def __sync(
```

Comment:
why not move this out of the LighningModule completely to a utilities file?

Reply:
Do you think other places might use it?
Do you have an idea of where it would fit better?

Reply:
I think it could be part of https://github.com/PyTorchLightning/pytorch-lightning/issues/7534

But this is definitely not a blocker for this refactoring

Reply:
Sure, feel free to include it when that's done